index_impl.go

```diff
@@ -530,10 +530,23 @@ func (i *indexImpl) SearchInContext(ctx context.Context, req *SearchRequest) (sr
 			} else if facetRequest.DateTimeRanges != nil {
 				// build date range facet
 				facetBuilder := facet.NewDateTimeFacetBuilder(facetRequest.Field, facetRequest.Size)
-				dateTimeParser := i.m.DateTimeParserNamed("")
 				for _, dr := range facetRequest.DateTimeRanges {
-					start, end := dr.ParseDates(dateTimeParser)
-					facetBuilder.AddRange(dr.Name, start, end)
+					dateTimeParserName := defaultDateTimeParser
+					if dr.DateTimeParser != "" {
+						dateTimeParserName = dr.DateTimeParser
+					}
+					dateTimeParser := i.m.DateTimeParserNamed(dateTimeParserName)
+					if dateTimeParser == nil {
+						return nil, fmt.Errorf("no date time parser named `%s` registered", dateTimeParserName)
```

Comment:
Wondering if the better thing to do here would skip this data point as opposed to failing the entire facets computations. This is applicable to 544 and 547 as well. Thoughts?

Reply:
but then the user will not be notified why there are no facet results right? Previous to this commit, the behaviour for ParseDates error was ambiguous, if i ran a query with malformed entries (datetime parser fails) on bleve directly the results were indeed dropped, i.e facet result did not have the corresponding date range. But running via couchbase, an error was thrown "start and end cannot be nil" for the same query. This was because the validate facet request() was called first, and then the query was executed. (validate facet request() throws that error on dateTimeParser failure).

So if validation() throws an error shouldn't running the actual query throw the same error too? The main goal also was to fix the error message thrown.

Reply:
Not always "no facet results" though. It could be a situation where some documents have malformed content while many/most others don't. This is entirely up to the user. I worry if we take this away - the change in behavior may not be taken lightly by some.

Reply:
but this parseDates function is to parse the start and end sate strings of the FacetQuery object, it is unrelated to document match contents.

Reply:
Ah I see. My mistake. This looks ok then 👍🏼